### PR TITLE
Added emplace method to AssociativeVector.

### LINKED ISSUE #256: Add `emplace` to AssociativeVector
and use it in `game_state_accumulator_state_register.cpp`

### PR DESCRIPTION
**Pull request by dino8890**
Fix for the issue #256.

### CODE REVIEW

File: src/xrCore/Containers/AssociativeVector.hpp [hunk 1]
```diff
@@ -302,6 +303,11 @@ IC void _associative_vector::insert(TIterator first, TIterator last)
     std::sort(begin(), end(), static_cast<TComparer&>(*this));
 }
 
+TEMPLATE_SPECIALIZATION
+IC typename _associative_vector::insert_result _associative_vector::emplace(key_type key, mapped_type value) {
+    return insert(std::make_pair(key, value));
```
Comment (Xottab-DUTY) [inline src/xrCore/Containers/AssociativeVector.hpp]: It should not call own `insert` function. It should call `emplace` from the underlying `xr_vector`.

Comment (dino8890) [reply src/xrCore/Containers/AssociativeVector.hpp]: Hey, thanks for the feedback!

Figured that might be the case, although it didn't seem prudent at first considering the vector elements can't ever be anything else than an `std::pair`. I will get back on it, and fix the issue.

**Comment by Xottab-DUTY** [comment]
Thank you for all the changes! Though, they are needed, they are not related to this pull request.

Please, move other changes into new pull request and leave in this pull request only changes that are related to AssociativeVector.

**Comment by dino8890** [comment]
Will do, cheers!

**Comment by dino8890** [comment]
This should be it, let me know if you need additional changes.

**Review by Xottab-DUTY — APPROVED** [review]

**Comment by Xottab-DUTY** [comment]
Thank you!

Sorry for the long review, I was sick and then was a bit busy with other stuff.. image

**Pull request merged**